File: internal/core/ports/repositories/patternconfig.go
/*
 * Copyright 2023-2025 Eric Moss
 * Licensed under FSL-1.1-ALv2 (Functional Source License 1.1, Apache 2.0 Future)
 * Full license: https://github.com/emoss08/Trenova/blob/master/LICENSE.md */

package repositories

import (
	"context"

	"github.com/emoss08/trenova/internal/core/domain/dedicatedlane"
	"github.com/emoss08/trenova/pkg/types/pulid"
)

type GetPatternConfigRequest struct {
	OrgID  pulid.ID
	BuID   pulid.ID
	UserID pulid.ID
}

type PatternConfigRepository interface {
	GetAll(ctx context.Context) ([]*dedicatedlane.PatternConfig, error)
	GetByOrgID(
		ctx context.Context,
		req GetPatternConfigRequest,
	) (*dedicatedlane.PatternConfig, error)
	Update(
		ctx context.Context,
		pc *dedicatedlane.PatternConfig,
	) (*dedicatedlane.PatternConfig, error)
}

File: ui/src/services/pattern-config.ts
/**
 * # Copyright 2023-2025 Eric Moss
 * # Licensed under FSL-1.1-ALv2 (Functional Source License 1.1, Apache 2.0 Future)
 * # Full license: https://github.com/emoss08/trenova/blob/main/LICENSE.md
 */

import { http } from "@/lib/http-client";
import type { PatternConfigSchema } from "@/lib/schemas/pattern-config-schema";

export class PatternConfigAPI {
  async get() {
    const response = await http.get<PatternConfigSchema>("/pattern-config/");
    return response.data;
  }

  async update(data: PatternConfigSchema) {
    const response = await http.put<PatternConfigSchema>(
      `/pattern-config/`,
      data,
    );
    return response.data;
  }
}
